plotly/tools.py

```diff
@@ -191,30 +198,46 @@ def set_config_file(plotly_domain=None,
     settings = get_config_file()
     if isinstance(plotly_domain, six.string_types):
         settings['plotly_domain'] = plotly_domain
+    elif plotly_domain is not None:
+        raise TypeError('Input should be a string')
```

Comment:
yes, i think this is fine. in the future, i'd like even more validation here, error-ing if someone includes `https://` in streaming domain